server/src/main/java/org/elasticsearch/common/time/FormatNames.java

```diff
@@ -122,7 +139,34 @@ public static boolean exist(String format) {
         return ALL_NAMES.contains(format);
     }
 
+    public static FormatNames forName(String format) {
+        for (FormatNames name : values()) {
+            if (name.matches(format)) {
+                return name;
+            }
+        }
+        throw new IllegalArgumentException("Format name " + format + " is not defined");
+    }
+
+
     public boolean matches(String format) {
-        return format.equals(camelCaseName) || format.equals(snakeCaseName);
+        if (format.equals(camelCaseName) || format.equals(snakeCaseName)) {
+            deprecate(format);
+            return true;
+        }
+        return false;
+    }
+
+    private void deprecate(String format) {
+        if (DEPRECATED_CAMEL_CASE_NAMES.contains(format)) {
```

Comment:
Shouldn't all camelCase names be deprecated, in which case we can only log the deprecated message if the first condition matches above?

Reply:
yes, all camel case should be deprecated. It is just for some like `iso8601`, `hour`, `year` etc snake case and camel case is the same. We do not need to log the deprecation for these

Reply:
In that case perhaps we could not fill in the camelCaseName and condition on that being non null? It seems like we are doing extra work to analyze the casing of the name when we have already separated the ok (snake case) from the deprecated (camelCase).

Reply:
I see what you mean now. That's the smart way to do it - will fix!